charmcraft/models/project.py

```diff
@@ -440,8 +441,18 @@ class CharmcraftProject(models.Project, metaclass=abc.ABCMeta):
     summary: CharmcraftSummaryStr | None
     description: str | None
 
-    analysis: AnalysisConfig | None
-    charmhub: CharmhubConfig | None
+    analysis: AnalysisConfig | None = pydantic.Field(
+        default=None,
+        description=textwrap.dedent(
```

Comment:
In this file, most of these field additions are to add the description and potentially examples. The description is used in the sphinx docs and in the JSON schema, but the examples are only made available in the JSON Schema right now. (This is a limitation of the pydantic sphinx plugin - I haven't looked into whether this is remedied in the pydantic 2 version.)

They're all copied from their relevant sections in https://juju.is/docs/sdk/charmcraft-yaml